src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -714,6 +716,60 @@ bool ZoneSet::CalculateGridZones(Rect workArea, JSONHelpers::GridLayoutInfo grid
     return success;
 }
 
+IFACEMETHODIMP_(bool)
+ZoneSet::SwitchFocusByDirection(HWND window, HWND windowZone, DWORD vkCode, bool cycle) noexcept
+{
+    winrt::com_ptr<IZone> zone;
+
+    if (m_zones.empty())
+    {
+        return false;
+    }
+
+    auto iter = std::find(m_zones.begin(), m_zones.end(), ZoneFromWindow(window));
+    if (iter == m_zones.end())
+    {
+        iter = (vkCode == VK_RIGHT) ? m_zones.end() - 1 : m_zones.begin();
```

Comment:
Is it possible that all zones are empty on start? If so I need to handle this corner case to avoid infinite loop.

Reply:
Yes, it's possible.